src/controllers/deleteGuildRoleHandler.ts

```diff
@@ -0,0 +1,43 @@
+import { IRequest } from "itty-router";
+import JSONResponse from "../utils/JsonResponse";
+import { env } from "../typeDefinitions/default.types";
+import * as response from "../constants/responses";
+import { verifyNodejsBackendAuthToken } from "../utils/verifyAuthToken";
+import { deleteGuildRole } from "../utils/deleteGuildRole";
+
+export async function deleteGuildRoleHandler(request: IRequest, env: env) {
+  const authHeader = request.headers.get("Authorization");
+  const reason = request.headers.get("X-Audit-Log-Reason");
+  const roleId = decodeURI(request.params?.roleId ?? "");
+  const { dev } = request.query;
+  const devFlag = dev === "true";
+
+  if (!devFlag) {
+    return new JSONResponse(response.NOT_FOUND, { status: 404 });
```

Comment:
Also shouldn't it be 501 Not Implemented as this request doesn't exist without a dev flag?